sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -71,4 +72,19 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
     assert(!rawTable.properties.contains(HiveExternalCatalog.DATASOURCE_PROVIDER))
     assert(DDLUtils.isHiveTable(externalCatalog.getTable("db1", "hive_tbl")))
   }
+
+  test("alter table schema should save case-sensitive schema to properties") {
```

Comment:
`HiveExternalCatalogSuite` extends `ExternalCatalogSuite`, so https://github.com/apache/spark/pull/16944/files#diff-8c4108666a6639034f0ddbfa075bcb37R243 will also be run here. Do we still need this test?